Fix protobuffer definitions

ManfredKarrer opened the pull request:
Deprecated PersistableNetworkPayloadList had initially the message type
number 12 and got changed to 15 which caused errors. We changed back so
that PersistableNetworkPayloadList has nr 12 again.
AccountAgeWitnessStore and TradeStatistics2Store also have been changed
to have the same numbers as on th latest release again.

Here are the items from the latest 0.7.1 release:
...
PersistableNetworkPayloadList persistable_network_payload_list = 12;
AccountAgeWitnessStore account_age_witness_store = 13;
TradeStatistics2Store trade_statistics2_store = 14;
State state = 15;
...

Here are the current items:
PersistableNetworkPayloadList persistable_network_payload_list = 12;
AccountAgeWitnessStore account_age_witness_store = 13;
TradeStatistics2Store trade_statistics2_store = 14;
BsqState bsq_state = 15;

ManfredKarrer committed:
Fix protobuffer definitions

Deprecated PersistableNetworkPayloadList had initially the message type
number 12 and got changed to 15 which caused errors. We changed back so
that PersistableNetworkPayloadList has nr 12 again.
AccountAgeWitnessStore and TradeStatistics2Store also have been changed
to have the same numbers as on th latest release again.

Here are the items from the latest 0.7.1 release:
...
PersistableNetworkPayloadList persistable_network_payload_list = 12;
AccountAgeWitnessStore account_age_witness_store = 13;
TradeStatistics2Store trade_statistics2_store = 14;
State state = 15;
...

Here are the current items:
PersistableNetworkPayloadList persistable_network_payload_list = 12;
AccountAgeWitnessStore account_age_witness_store = 13;
TradeStatistics2Store trade_statistics2_store = 14;
BsqState bsq_state = 15;

ripcurlx commented:
Unfortunately still fails for me.
<img width="1284" alt="bildschirmfoto 2018-08-20 um 19 02 36" src="https://user-images.githubusercontent.com/170962/44354996-9e7aa200-a4ab-11e8-9f81-236e3387d8e0.png">

A little bit different exception, but still throws an error on startup.
```
Aug-20 19:00:52.795 [readFromResourcesThread] ERROR b.c.s.CommonSetup: Uncaught Exception from thread readFromResourcesThread 
Aug-20 19:00:52.795 [readFromResourcesThread] ERROR b.c.s.CommonSetup: throwableMessage= bisq.core.dao.state.BsqState cannot be cast to bisq.network.p2p.storage.persistence.PersistableNetworkPayloadList 
Aug-20 19:00:52.795 [readFromResourcesThread] ERROR b.c.s.CommonSetup: throwableClass= class java.lang.ClassCastException 
java.lang.ClassCastException: bisq.core.dao.state.BsqState cannot be cast to bisq.network.p2p.storage.persistence.PersistableNetworkPayloadList
	at bisq.network.p2p.storage.persistence.PersistableNetworkPayloadListService.getMap(PersistableNetworkPayloadListService.java:103)
	at bisq.network.p2p.storage.persistence.AppendOnlyDataStoreService.transferDeprecatedDataStructure(AppendOnlyDataStoreService.java:66)
	at bisq.network.p2p.storage.persistence.AppendOnlyDataStoreService.readFromResources(AppendOnlyDataStoreService.java:59)
	at bisq.network.p2p.storage.P2PDataStorage.readFromResources(P2PDataStorage.java:155)
	at bisq.core.app.SetupUtils.lambda$readFromResources$1(SetupUtils.java:98)
	at java.lang.Thread.run(Thread.java:748)
Aug-20 19:00:52.798 [readFromResourcesThread] ERROR b.c.s.CommonSetup: Stack trace:
java.lang.ClassCastException: bisq.core.dao.state.BsqState cannot be cast to bisq.network.p2p.storage.persistence.PersistableNetworkPayloadList
	at bisq.network.p2p.storage.persistence.PersistableNetworkPayloadListService.getMap(PersistableNetworkPayloadListService.java:103)
	at bisq.network.p2p.storage.persistence.AppendOnlyDataStoreService.transferDeprecatedDataStructure(AppendOnlyDataStoreService.java:66)
	at bisq.network.p2p.storage.persistence.AppendOnlyDataStoreService.readFromResources(AppendOnlyDataStoreService.java:59)
	at bisq.network.p2p.storage.P2PDataStorage.readFromResources(P2PDataStorage.java:155)
	at bisq.core.app.SetupUtils.lambda$readFromResources$1(SetupUtils.java:98)
	at java.lang.Thread.run(Thread.java:748)
```

ManfredKarrer committed:
Fix previous commit for protobuffer definitions

PersistableNetworkPayloadList had message number 15 in v0.7.1 not as
thought in the previous commit 12

ManfredKarrer committed:
Add removeAndBackupFile method

ripcurlx reviewed (approved):
ACK
